### PR TITLE
Video multi qualities

### PR DESCRIPTION
**Pull request by bbaudel-gpsw**
Update the multi-qualities managment for Firefox, Edge, IE11 compatibility.

Fixes :
- solve the frequent timeout event on MS browsers when requesting a quality
- second try added when a video quality request is aborted due to timeout
- add a downgrade counter to force the quality change only if the video is really "freezed"

Feature :
- add a startTime option for the media into JSON configuration to be able to set the currentTime of the video on start

### CODE REVIEW

**Comment by sw-team-release-gpsw** [comment]
[CLA assistant check](https://cla.gopro.com/sign) <br/>All committers have signed the CLA.

File: src/display/video/VideoHTML5.js [hunk 1]
```diff
@@ -1765,9 +1804,9 @@ FORGE.VideoHTML5.prototype._onEventHandler = function(event)
             break;
 
         case "volumechange":
-            // I do not dispatch the volume change if readyState is 0. Because
-            // I set the volume at 0 when I create the video element, it is
-            // not usefull to dispatch this internal volume change ?
+            //I do not dispatch the volume change if readyState is 0. Because
+            //I set the volume at 0 when I create the video element, it is
+            //not usefull to dispatch this internal volume change ?
             if (this._onVolumeChange !== null && element.readyState !== 0)
```
Comment (aberthet-gpsw) [inline src/display/video/VideoHTML5.js]: While you're at it, can you change this to `element.readyState !== HTMLMediaElement.HAVE_NOTHING` ? Same above with `durationchange` and `HTMLMediaElement.HAVE_METADATA`.

**Comment by aberthet-gpsw** [comment]
All tests are goods to me !

**Comment by aberthet-gpsw** [comment]
Good to merge !

**Comment by bbaudel-gpsw** [comment]
Add devices detection for video source levels urls.

Usage
`"media":
{
	"uid": "media-video",
	"type": "video",
	"source":
	{
		"format": "equi",
		"levels": [
			{
				"url": "720p.mp4",
				"device": { "mobile": false }
			},
			{ "url": "1080p.mp4" },
			{
				"url": "1440p.mp4",
				"device": { "mobile": false }
			},
			{
				"url": "2160p.mp4",
				"device": { "safari": false, "firefox": false, "mobile": false }
			}
		]
	}
}`

**Pull request merged**